server/hotrod/src/main/scala/org/infinispan/server/hotrod/Decoder2x.scala

```diff
@@ -585,7 +604,7 @@ object Decoder2x extends AbstractVersionedDecoder with ServerConstants with Log
     * Convert an expiration value into milliseconds
     */
    override def toMillis(param: ExpirationParam, h: SuitableHeader): Long = {
-      if (h.version == VERSION_22) {
+      if (h.version == VERSION_22 | h.version == VERSION_23) {
```

Comment:
On a second thought, this refactor would bring a bit of noise to this PR, I've added a comment on https://issues.jboss.org/browse/ISPN-5342 to address this separation together with the other refactorings